diff --git a/Carthage/Checkouts/Magnet/Lib/Magnet/HotKeyCenter.swift b/Carthage/Checkouts/Magnet/Lib/Magnet/HotKeyCenter.swift
index 167f77d..2facefb 100644
--- a/Carthage/Checkouts/Magnet/Lib/Magnet/HotKeyCenter.swift
+++ b/Carthage/Checkouts/Magnet/Lib/Magnet/HotKeyCenter.swift
@@ -113,19 +113,6 @@ private extension HotKeyCenter {
         InstallEventHandler(GetEventDispatcherTarget(), { (_, inEvent, _) -> OSStatus in
             return HotKeyCenter.shared.sendCarbonEvent(inEvent!)
         }, 1, &pressedEventType, nil, nil)
-
-        // Press Modifiers Event
-        let mask = CGEventMask((1 << CGEventType.flagsChanged.rawValue))
-        let event = CGEvent.tapCreate(tap: .cghidEventTap,
-                                     place: .headInsertEventTap,
-                                     options: .listenOnly,
-                                     eventsOfInterest: mask,
-                                     callback: { (_, _, event, _) in return HotKeyCenter.shared.sendModifiersEvent(event) },
-                                     userInfo: nil)
-        if event == nil { return }
-        let source = CFMachPortCreateRunLoopSource(kCFAllocatorDefault, event!, 0)
-        CFRunLoopAddSource(CFRunLoopGetCurrent(), source, CFRunLoopMode.commonModes)
-        CGEvent.tapEnable(tap: event!, enable: true)
     }
 
     func sendCarbonEvent(_ event: EventRef) -> OSStatus {
